### PR TITLE
Warn AMO users when visiting the site with Fenix

### LINKED ISSUE #9068: Warn AMO users when visiting the site with Fenix
From the PRD at https://docs.google.com/document/d/1HOImPu2RO9PXIbrNvE_YhYyONw7n-rgjzMtMQ3Knk4Q/edit.

When a user visits AMO using Fenix, a warning should be displayed similar to the platform warning implemented via #8701. An an add-on detail page, the warning will appear below the install button, similar to the current warning:

Screenshot 2019-10-21 12 20 03

The text for this warning will be "This add-on is not compatible with Firefox for Android. Learn More."

On all other pages, the warning will appear at the top of the page, similar to the current warning:

Screenshot 2019-10-21 12 19 47

The text for this warning will be "To learn about add-ons compatible with Firefox for Android, click here."

In each case there will be a link to either a SUMO article or a collection of add-ons that are compatible with Fenix. That is TBD.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #9068 

This PR introduces a few changes:

1. The `AddonCompatibilityError`, which is the red warning near the top of the add-on detail card, will no longer be displayed when a user visits the site with Fenix. This is because that warning is being replaced by the `WrongPlatformWarning`. Before screenshot:

Screenshot 2020-01-13 12 09 58

2. When visiting the site with Fenix, the `WrongPlatformWarning` will be displayed with a message. This will either be displayed under the install button (on the add-on detail page):

Screenshot 2020-01-13 12 07 39

or at the top of the screen:

Screenshot 2020-01-13 12 08 00

3. Similar to 2, above, the standard `WrongPlatformWarning`, which displays a warning if you visit the `/android` site with desktop, or the `/firefox` site with Android, will *not* be displayed for Fenix users, as it will be replaced by the new `WrongPlatformWarning` as described in 2, above.

### CODE REVIEW

**Comment by bobsilverberg** [comment]
Well, I tried to use mozilla/addons-engineering for this assignment, but it just chose @willdurand anyway. 😜

**Comment by bobsilverberg** [comment]
Note that we are waiting on a final SUMO link for this, so we should not merge this until we have that. I will update the milestone if we don't have a proper link by tomorrow.

**Comment by bobsilverberg** [comment]
@caitmuenster Please let me know once we have a correct link for SUMO for this.

File: src/amo/components/WrongPlatformWarning/index.js [hunk 1]
```diff
@@ -35,46 +43,61 @@ type InternalProps = {|
...
+    let message;
 
-    if (!newLocation) {
-      return null;
-    }
+    if (_isFenix(userAgentInfo)) {
+      message = i18n.sprintf(
+        this.props.fixFenixLinkMessage ||
+          i18n.gettext(
+            `To find add-ons compatible with Firefox on Android,
```
Comment (willdurand) [inline src/amo/components/WrongPlatformWarning/index.js]: I think it is `Firefox for Android`

Comment (bobsilverberg) [reply src/amo/components/WrongPlatformWarning/index.js]: Indeed! That's also what's in the doc at https://docs.google.com/document/d/1HOImPu2RO9PXIbrNvE_YhYyONw7n-rgjzMtMQ3Knk4Q/edit?ts=5e18f98a#heading=h.lzrviwt2ql63. I have no idea where that "on" came from!

**Comment by caitmuenster** [comment]
Thanks @bobsilverberg! The article is published at https://support.mozilla.org/kb/add-compatibility-firefox-preview/.

**Comment by bobsilverberg** [comment]
Thanks @willdurand and @caitmuenster. I addressed the review comment and have updated the SUMO link. I still have an open question for @jvillalobos at https://github.com/mozilla/addons-frontend/issues/9068#issuecomment-574193847 about the wording for the warnings, but that could be a follow-up.

**Pull request merged**